### ISSUE TITLE
Disable CA1305 for Guid.ToString

### ISSUE THREAD

**Issue by msmolka** [dotnet/roslyn-analyzers#3378]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.8 (Latest)

#### Diagnostic ID

CA1305: Specify IFormatProvider

#### Repro steps

```c#
Guid a = Guid.NewGuid();
string test = a.ToSrring("D");
```

#### Expected behavior

`a.ToString` is not underlined by code analyzer

#### Actual behavior
`a.ToString` is underlined by code analyzer

as docs:
[https://docs.microsoft.com/en-US/dotnet/api/system.guid.tostring?view=netframework-4.8#System_Guid_ToString_System_String_System_IFormatProvider_](https://docs.microsoft.com/en-US/dotnet/api/system.guid.tostring?view=netframework-4.8#System_Guid_ToString_System_String_System_IFormatProvider_)

> Because the provider parameter is ignored, you cannot use it to provide a custom formatting solution. To represent a Guid value as a string in a format that isn't supported by the standard GUID format strings, call the String.Format(IFormatProvider, String, Object[]) method with a provider object that implements both the ICustomFormatter and IFormatProvider interfaces. For more information, see the "Custom Formatting with ICustomFormatter" section in the Formatting Types article.

It is similar as https://github.com/dotnet/roslyn-analyzers/issues/2394

**Comment by sharwell** [comment]
This sentence of the documentation makes me slightly uneasy, but if @stephentoub doesn't think it will change then would could make a global exclusion for this rule. (Emphasis mine)

> The `provider` parameter **is reserved for future use** and does not contribute to the execution of this method.

**Comment by Evangelink** [comment]
Don't you think that would still be valuable to exclude Guids until the provider is actually used in the method? At this point, we could remove the exclusion and update the docs.

**Comment by stephentoub** [comment]
I'm fine excluding it; I don't see us ever changing the implementation to factor it in, at least not for existing format characters, and if we did ever start respecting it, we could update the rule at that time presumably.

Note, though, that it's not the only Guid that ignores it, e.g. `char.ToString` also ignores the provider, as does `Enum.ToString`, maybe some others I can't think of right now.
